### ISSUE TITLE
Value mappings with regex not working

### ISSUE THREAD

**Issue by lidora-everc** [grafana/grafana#72970]
### What happened?

I have `grafana v9.3.1` and value mappings with regex not working,
The dashboard json as following:
| Key | Value |
|--|--|
| Panel | stat @ 9.3.1 |
| Grafana | 9.3.1 (89b365f8b1) // Open Source |
<details><summary>Panel debug snapshot dashboard</summary>

```json
{
  "panels": [
    {
      "id": 2,
      "gridPos": {
        "h": 13,
        "w": 15,
        "x": 0,
        "y": 0
      },
      "type": "stat",
      "title": "Reproduced with embedded data",
      "transformations": [],
      "datasource": {
        "type": "grafana",
        "uid": "grafana"
      },
      "pluginVersion": "9.3.1",
      "fieldConfig": {
        "defaults": {
          "mappings": [
            {
              "type": "regex",
              "options": {
                "pattern": "(.*)",
                "result": {
                  "text": "$1%",
                  "index": 0
                }
              }
            }
          ],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              }
            ]
          },
          "color": {
            "mode": "thresholds"
          },
          "unit": "string"
        },
        "overrides": []
      },
      "options": {
        "reduceOptions": {
          "values": false,
          "calcs": [
            "lastNotNull"
          ],
          "fields": ""
        },
        "orientation": "auto",
        "textMode": "auto",
        "colorMode": "value",
        "graphMode": "area",
        "justifyMode": "auto"
      },
      "targets": [
        {
          "refId": "A",
          "datasource": {
            "type": "grafana",
            "uid": "grafana"
          },
          "queryType": "snapshot",
          "snapshot": [
            {
              "schema": {
                "refId": "A",
                "meta": {
                  "type": "timeseries-many",
                  "custom": {
                    "resultType": "matrix"
                  },
                  "executedQueryString": "Expr: (sum(sum_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*\"}[15634800s]) unless sum_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*metrics.*\"}[15634800s])) / \nsum(count_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*\"}[15634800s]) unless count_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*metrics.*\"}[15634800s]))) * 100\nStep: 6h0m0s",
                  "preferredVisualisationType": "graph"
                },
                "name": "(sum(sum_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*\"}[15634800s]) unless sum_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*metrics.*\"}[15634800s])) / \nsum(count_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*\"}[15634800s]) unless count_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*metrics.*\"}[15634800s]))) * 100",
                "fields": [
                  {
                    "name": "Time",
                    "type": "time",
                    "typeInfo": {
                      "frame": "time.Time"
                    },
                    "config": {
                      "interval": 21600000
                    }
                  },
                  {
                    "name": "Value",
                    "type": "number",
                    "typeInfo": {
                      "frame": "float64"
                    },
                    "labels": {},
                    "config": {
                      "displayNameFromDS": "(sum(sum_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*\"}[15634800s]) unless sum_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*metrics.*\"}[15634800s])) / \nsum(count_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*\"}[15634800s]) unless count_over_time(kube_pod_status_phase{namespace=\"keda\",phase=\"Running\",pod=~\"^keda-operator-.*metrics.*\"}[15634800s]))) * 100"
                    }
                  }
                ]
              },
              "data": {
                "values": [
                  [
                    1688180400000,
                    1688202000000,
                    1688223600000,
                    1688245200000,
                    1688266800000,
                    1688288400000,
                    1688310000000,
                    1688331600000,
                    1688353200000,
                    1688374800000,
                    1688396400000,
                    1688418000000,
                    1688439600000,
                    1688461200000,
                    1688482800000,
                    1688504400000,
                    1688526000000,
                    1688547600000,
                    1688569200000,
                    1688590800000,
                    1688612400000,
                    1688634000000,
                    1688655600000,
                    1688677200000,
                    1688698800000,
                    1688720400000,
                    1688742000000,
                    1688763600000,
                    1688785200000,
                    1688806800000,
                    1688828400000,
                    1688850000000,
                    1688871600000,
                    1688893200000,
                    1688914800000,
                    1688936400000,
                    1688958000000,
                    1688979600000,
                    1689001200000,
                    1689022800000,
                    1689044400000,
                    1689066000000,
                    1689087600000,
                    1689109200000,
                    1689130800000,
                    1689152400000,
                    1689174000000,
                    1689195600000,
                    1689217200000,
                    1689238800000,
                    1689260400000,
                    1689282000000,
                    1689303600000,
                    1689325200000,
                    1689346800000,
                    1689368400000,
                    1689390000000,
                    1689411600000,
                    1689433200000,
                    1689454800000,
                    1689476400000,
                    1689498000000,
                    1689519600000,
                    1689541200000,
                    1689562800000,
                    1689584400000,
                    1689606000000,
                    1689627600000,
                    1689649200000,
                    1689670800000,
                    1689692400000,
                    1689714000000,
                    1689735600000,
                    1689757200000,
                    1689778800000,
                    1689800400000,
                    1689822000000,
                    1689843600000,
                    1689865200000,
                    1689886800000,
                    1689908400000,
                    1689930000000,
                    1689951600000,
                    1689973200000,
                    1689994800000,
                    1690016400000,
                    1690038000000,
                    1690059600000,
                    1690081200000,
                    1690102800000,
                    1690124400000,
                    1690146000000,
                    1690167600000,
                    1690189200000,
                    1690210800000,
                    1690232400000,
                    1690254000000,
                    1690275600000,
                    1690297200000,
                    1690318800000,
                    1690340400000,
                    1690362000000,
                    1690383600000,
                    1690405200000,
                    1690426800000,
                    1690448400000,
                    1690470000000,
                    1690491600000,
                    1690513200000,
                    1690534800000,
                    1690556400000,
                    1690578000000,
                    1690599600000,
                    1690621200000,
                    1690642800000,
                    1690664400000,
                    1690686000000,
                    1690707600000,
                    1690729200000,
                    1690750800000,
                    1690772400000,
                    1690794000000,
                    1690815600000,
                    1690837200000,
                    1690858800000,
                    1690880400000,
                    1690902000000,
                    1690923600000,
                    1690945200000,
                    1690966800000,
                    1690988400000,
                    1691010000000,
                    1691031600000,
                    1691053200000,
                    1691074800000,
                    1691096400000,
                    1691118000000,
                    1691139600000,
                    1691161200000,
                    1691182800000,
                    1691204400000,
                    1691226000000,
                    1691247600000,
                    1691269200000,
                    1691290800000
                  ],
                  [
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    100,
                    99.99170227772477,
                    99.99193905928821,
                    99.99216638596216,
                    99.99238124261933,
                    99.99258737630184,
                    99.99278004404172,
                    99.98943885094698,
                    99.98969850971774,
                    99.98994638069705,
                    99.99018099695611,
                    99.99040706040354,
                    99.99062294877005,
                    99.99082905355833,
                    99.99102656137832,
                    99.99121574139143,
                    99.99139661600229,
                    99.9915706659174,
                    99.99173667538906,
                    99.99189736664415,
                    99.9920519274076,
                    99.99220070193682,
                    99.99234400918718,
                    99.9924817682881,
                    99.9926142938034,
                    99.9927429304047,
                    99.99286716279512,
                    99.99298721335235,
                    99.9931032897308,
                    99.99321558606029,
                    99.99332383834786,
                    99.99342897820611,
                    99.99353099730457,
                    99.99362949120871,
                    99.99372542457961,
                    99.99381851149757,
                    99.99390887679688,
                    99.99399627769216,
                    99.99408155615616,
                    99.99416444591414,
                    99.9942447147297,
                    99.99432312757824,
                    99.99439943247582,
                    99.99447371329624,
                    99.99454604952187,
                    99.99461651652729,
                    99.99468518584133,
                    99.99475212538923,
                    99.99481739971668,
                    99.9948810701976,
                    99.99494319522637,
                    99.9950037471896,
                    99.99506286617076,
                    99.9951206818034,
                    99.99517715902513,
                    99.99523234377979,
                    99.99528627993213,
                    99.99533900938414,
                    99.99539057218364,
                    99.99544100662574,
                    99.9954903493476,
                    99.99553863541729,
                    99.9955858984168,
                    99.99563217052007,
                    99.99567748256585,
                    99.99572186412641,
                    99.99576534357179,
                    99.99580794813035,
                    99.99584947426673,
                    99.99589041095889,
                    99.99593060322024,
                    99.99596990865126,
                    99.99600851516765,
                    99.99604638903531,
                    99.99608355091384,
                    99.99612002069323,
                    99.99615581752947,
                    99.9961909598781,
                    99.99622546552592,
                    99.99625935162095,
                    99.99629249725027,
                    99.99632519568334,
                    99.9963573223891,
                    99.9963888922325,
                    99.99641991956753,
                    99.99645041825906,
                    99.99648040170348,
                    99.9965098828484,
                    99.99653887421115,
                    99.9965673878966,
                    99.9965946240465,
                    99.99662139333739,
                    99.99664856893894,
                    99.99667505292209,
                    99.99670137552641,
                    99.99672724892544,
                    99.99675257899352,
                    99.99677769304304,
                    99.99680238755063,
                    99.99682674000422,
                    99.99685072433341,
                    99.99687434882267,
                    99.99689762150983,
                    99.99692051858467,
                    99.99694308015243,
                    99.99696531353369,
                    99.99698725609328,
                    99.99700888361566,
                    99.99703020283715,
                    99.99705122030333,
                    99.99707194237583,
                    99.99709237523867,
                    99.99711252490448
                  ]
                ]
              }
            }
          ]
        }
      ]
    },
    {
      "gridPos": {
        "h": 7,
        "w": 9,
        "x": 15,
        "y": 0
      },
      "id": 5,
      "options": {
        "content": "<table width=\"100%\">\n    <tr>\n      <th width=\"2%\">Panel</th>\n      <td >stat @ 9.3.1</td>\n    </tr>\n    <tr>\n      <th>Queries</th>\n      <td>A[prometheus]</td>\n    </tr>\n    <tr>\n      <th>Transforms (0)</th>\n      <td></td>\n  </tr>\n    <tr><th>Data</th><td> 1 frames, 2 fields, 145 rows </td></tr>\n    \n    <tr>\n      <th>Grafana</th>\n      <td>9.3.1 (89b365f8b1) // Open Source</td>\n    </tr>\n  </table>",
        "mode": "html"
      },
      "title": "Debug info",
      "type": "text"
    },
    {
      "id": 6,
      "title": "Original Panel JSON",
      "type": "text",
      "gridPos": {
        "h": 13,
        "w": 9,
        "x": 15,
        "y": 7
      },
      "options": {
        "content": "{\n  \"id\": 8,\n  \"gridPos\": {\n    \"h\": 8,\n    \"w\": 11,\n    \"x\": 0,\n    \"y\": 0\n  },\n  \"type\": \"stat\",\n  \"title\": \"Keda operator SLA in percentage \",\n  \"transformations\": [],\n  \"datasource\": {\n    \"uid\": \"prometheus\",\n    \"type\": \"prometheus\"\n  },\n  \"pluginVersion\": \"9.3.1\",\n  \"fieldConfig\": {\n    \"defaults\": {\n      \"mappings\": [\n        {\n          \"type\": \"regex\",\n          \"options\": {\n            \"pattern\": \"(.*)\",\n            \"result\": {\n              \"text\": \"$1%\",\n              \"index\": 0\n            }\n          }\n        }\n      ],\n      \"thresholds\": {\n        \"mode\": \"absolute\",\n        \"steps\": [\n          {\n            \"color\": \"green\",\n            \"value\": null\n          }\n        ]\n      },\n      \"color\": {\n        \"mode\": \"thresholds\"\n      },\n      \"unit\": \"string\"\n    },\n    \"overrides\": []\n  },\n  \"options\": {\n    \"reduceOptions\": {\n      \"values\": false,\n      \"calcs\": [\n        \"lastNotNull\"\n      ],\n      \"fields\": \"\"\n    },\n    \"orientation\": \"auto\",\n    \"textMode\": \"auto\",\n    \"colorMode\": \"value\",\n    \"graphMode\": \"area\",\n    \"justifyMode\": \"auto\"\n  },\n  \"targets\": [\n    {\n      \"datasource\": {\n        \"type\": \"prometheus\",\n        \"uid\": \"prometheus\"\n      },\n      \"editorMode\": \"code\",\n      \"expr\": \"(sum(sum_over_time(kube_pod_status_phase{namespace=\\\"keda\\\",phase=\\\"Running\\\",pod=~\\\"^keda-operator-.*\\\"}[$__range]) unless sum_over_time(kube_pod_status_phase{namespace=\\\"keda\\\",phase=\\\"Running\\\",pod=~\\\"^keda-operator-.*metrics.*\\\"}[$__range])) / \\nsum(count_over_time(kube_pod_status_phase{namespace=\\\"keda\\\",phase=\\\"Running\\\",pod=~\\\"^keda-operator-.*\\\"}[$__range]) unless count_over_time(kube_pod_status_phase{namespace=\\\"keda\\\",phase=\\\"Running\\\",pod=~\\\"^keda-operator-.*metrics.*\\\"}[$__range]))) * 100\",\n      \"legendFormat\": \"__auto\",\n      \"range\": true,\n      \"refId\": \"A\"\n    }\n  ]\n}",
        "mode": "code",
        "code": {
          "language": "json",
          "showLineNumbers": true,
          "showMiniMap": true
        }
      }
    },
    {
      "id": 3,
      "title": "Data from panel above",
      "type": "table",
      "datasource": {
        "type": "datasource",
        "uid": "-- Dashboard --"
      },
      "gridPos": {
        "h": 7,
        "w": 15,
        "x": 0,
        "y": 13
      },
      "options": {
        "showTypeIcons": true
      },
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "-- Dashboard --"
          },
          "panelId": 2,
          "withTransforms": true,
          "refId": "A"
        }
      ]
    }
  ],
  "schemaVersion": 37,
  "title": "Debug: Keda operator SLA in percentage  // 2023-08-06 11:11:25",
  "tags": [
    "debug",
    "debug-stat"
  ],
  "time": {
    "from": "2023-02-06T09:11:25.666Z",
    "to": "2023-08-06T08:11:25.666Z"
  }
}
```
</details>
and value mapping with regex not working.

### What did you expect to happen?

I using the following value mappings in the dashboard panel:
![image](https://github.com/grafana/grafana/assets/114167519/9ad387bf-017c-4970-86eb-0582564e8204)
And i expecting that the percentage symbol will be added to the numbers in the dashboard panel:
![image](https://github.com/grafana/grafana/assets/114167519/e634d2fe-4bf0-416a-8403-6d9b0d4b7b98)
But not occuring anything.

### Did this work before?

**Basically, this is the first time that i tested it.**

### How do we reproduce it?

- Trying to using value mappings with regex in another panel

- Trying to create same dashboard panel on another grafana with same version(including github sha) and the behaviour is the same

### Environment (with versions)?

Grafana: `v9.3.1 89b365f8b1`
OS: `Alpine Linux v3.15`
Browser: `Version 115.0.5790.114 (Official Build) (arm64)`

### Grafana platform?

Kubernetes

### Datasource(s)?

Prometheus
